### PR TITLE
Update crepr.py #51

### PR DESCRIPTION
**Pull request by Sohammhatre10**
### **User description**
Summary
This pull request introduces a new command report_missing t which reports classes in specified Python files that do not have a custom __repr__ method defined mentioned in https://github.com/cleder/crepr/issues/51

Changes Made
Command Definition: A new command report_missing has been added, which takes a list of file paths as input.
Module Inspection: The command iterates through the provided files, attempts to load each module, and checks for classes that either lack a __repr__ method or use the default implementation from the object class.
Error handling using CreprError.
Output: Classes missing a custom __repr__ method are reported with their file paths and line numbers for easy identification.
Motivation
Having a custom __repr__ method improves the readability and debugging of classes by providing meaningful string representations. This command helps developers ensure that their classes follow this best practice.

How to Test
Run the command in the terminal with a specified Python file as follows:
```python
!crepr report-missing <your-filename>
```

Expected output is of the form
``` 
file-name : line-number : class-name 
```

___

### **PR Type**
enhancement, error handling

___

### **Description**
- Introduced a new command `report_missing` to identify classes in specified Python files that do not have a custom `__repr__` method.
- Implemented error handling using `CreprError` to manage exceptions during module inspection.
- Added `report_missing_classes` function to iterate through modules and report classes missing a `__repr__` method with their file paths and line numbers.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>crepr.py</strong><dd><code>Add command to report missing __repr__ methods in classes</code></dd></summary>
<hr>

crepr/crepr.py

<li>Added a new command <code>report_missing</code> to identify classes without a <br>custom <code>__repr__</code> method.<br> <li> Implemented error handling using <code>CreprError</code> for the new command.<br> <li> Introduced <code>report_missing_classes</code> function to inspect modules and <br>report missing <code>__repr__</code> methods.<br>

</details>

  </td>
  <td><a href="https://github.com/cleder/crepr/pull/63/files#diff-abcd7c4a2773c8234cdc05070d0abd10b0245a1a15f470288240c6d3a108f576">+27/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

## Summary by Sourcery

Add a new command 'report_missing' to the crepr tool, which reports classes without a custom __repr__ method in specified Python files, enhancing code readability and debugging.

New Features:
- Introduce a new command 'report_missing' that identifies classes in specified Python files lacking a custom __repr__ method.

## Summary by CodeRabbit

- **New Features**
	- Introduced a `report_missing` command to identify classes lacking a `__repr__` method.
	- Users can input file paths to check for classes without a defined `__repr__`.

- **Bug Fixes**
	- Enhanced test coverage for the `report_missing` command with new test cases for various scenarios.

- **Documentation**
	- Added docstrings for the new command and helper function to clarify their purpose and usage.

### CODE REVIEW

**Comment by Sohammhatre10** [comment]
Hello @cleder,
Codecov is flagging lines 390, 402, and 404 in crepr/crepr.py as not covered by tests, leading to the coverage error. Could you provide some guidance on the best approach to resolve this?

**Comment by cleder** [comment]
Run `pytest  --cov=crepr --cov-report=term-missing` to produce the test coverage report [see also](https://github.com/cleder/pygeoif/issues/222#issuecomment-2379506205)

File: crepr/crepr.py [hunk 1]
```diff
@@ -377,5 +377,32 @@ def remove(
                 f.write("\n".join(src))
 
 
+@app.command()
+def report_missing(
+    files: Annotated[list[pathlib.Path], file_arg],
+) -> None:
+    """Report classes without __repr__ methods."""
+    for file_path in files:
```
Comment (cleder) [inline crepr/crepr.py:385]:
```suggestion
    for module, file_path in get_modules(files):
```
I don't think we need to reinvent the wheel here

File: crepr/crepr.py [hunk 2]
```diff
@@ -377,5 +377,32 @@ def remove(
...
+        try:
+            module = get_module(file_path)
+            report_missing_classes(module, file_path)
+        except CreprError as e:
```
Comment (cleder) [inline crepr/crepr.py:389]: not necessary if `get_modules` is used.

Comment (Sohammhatre10) [reply crepr/crepr.py:389]: Sure will make the changes, wanted ```get_module`` in the try block for covering any unexpected errors or return types from the ``get_module``` function, apologies for the oversight about the return type being defined as an Iterator for ```get_modules```
.

**Comment by Sohammhatre10** [comment]
1. Have added tests for ```report-missing``` and ```report_missing_classes``` functions along with a check for ```CreprError``` error handle call.
2. Have implemented the changes mentioned here https://github.com/cleder/crepr/pull/63#discussion_r1784275522

**Comment by Sohammhatre10** [comment]
@cleder Apologies for the oversight here had mixed up the `class_with_repr.py` test file and `class_without_repr.py` test file which cause cpython 3.12 to fail

**Review by cleder — APPROVED** [review]

**Comment by Sohammhatre10** [comment]
@cleder Will you please add the hacktoberfest-accepted tag on this.
Thanks!

**Pull request merged**